src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java

```diff
@@ -103,10 +104,24 @@ private ExtImpSovrn parseExtImpSovrn(ObjectNode ext) {
         }
     }
 
-    private static BigDecimal resolveBidFloor(BigDecimal impBidFloor, BigDecimal extBidFloor) {
-        return !BidderUtil.isValidPrice(impBidFloor) && BidderUtil.isValidPrice(extBidFloor)
-                ? extBidFloor
-                : impBidFloor;
+    private static BigDecimal resolveBidFloor(BigDecimal impBidFloor, JsonNode extBidFloor) {
+        final BigDecimal fromExt = parseBidFloor(extBidFloor);
+        return !BidderUtil.isValidPrice(impBidFloor) && BidderUtil.isValidPrice(fromExt) ? fromExt : impBidFloor;
+    }
+
+    private static BigDecimal parseBidFloor(JsonNode extBidFloor) {
+        if (extBidFloor != null) {
+            if (extBidFloor.isNumber()) {
+                return extBidFloor.decimalValue();
+            } else if (extBidFloor.isTextual()) {
+                try {
+                    return new BigDecimal(extBidFloor.textValue());
+                } catch (NumberFormatException e) {
+                    return BigDecimal.ZERO;
+                }
+            }
+        }
+        return BigDecimal.ZERO;
```

Comment:
you can revert this entirely as well as reverting the type change in the ExtImpSovrn, since both a string and a number will be successfully deserialized to BigDecimal

the only change you need is changing the json scheme, everything else should work fine

in order to test whether it works you can play with the integration test `SovrnTest` by changing values in the request json to double-check it works as intended

Reply:
thank you for making it simpler.